### DISCUSSION TITLE
some question with get_ref and get_ptr

### DISCUSSION THREAD

**Discussion by zhnkc9** [nlohmann/json#4185] [General]
hey guy . 

I want to know the difference between ptr and ref

in my code : 

auto arr = json.get_ref<json::array_t &>()
 
here , i update the arr ,but origin json data is not change 

but i use  get_ptr<json::array_t *>()

it will work ; 

can you tell me why bro . help me plz

**Comment by nlohmann** [comment]
use auto&

**Reply by zhnkc9** [reply]
thank you bro ,it worked !
